src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,130 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    static final int DEFAULT_CAPACITY = 16;
+    static final float LOAD_FACTOR = 0.75f;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        this.table = (Node<K, V>[]) new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if ((float) size / table.length >= LOAD_FACTOR) {
+            resize();
+        }
+        Node<K, V> newNode = new Node<>(key, value, null);
+        int index = getKeyHashcode(key) != 0
+                ? Math.abs(getKeyHashcode(newNode.key)) % table.length : 0;
```

Comment:
lets create a method that calculates the index

Reply:
Done